modules/ingester/instance.go

```diff
@@ -154,14 +144,39 @@ func newInstance(instanceID string, limiter *Limiter, overrides ingesterOverride
 	return i, nil
 }
 
-func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) error {
+func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) *tempopb.PushResponse {
+	atLeastOneError := false
+	pr := &tempopb.PushResponse{}
+
 	for j := range req.Traces {
+
 		err := i.PushBytes(ctx, req.Ids[j].Slice, req.Traces[j].Slice)
 		if err != nil {
-			return err
+			// only make list if there is at least one error
+			if !atLeastOneError {
```

Comment:
let's make this block a function for clarity. we can also test it directly which is nice b/c it does some weird things.

something like:
```
  addTraceError([]tempopb.PushErrorReason, pos int, err error) []tempopb.PushErrorReason
```